src/spikeinterface_pipelines/preprocessing/params.py

```diff
@@ -41,12 +41,125 @@ class HighpassSpatialFilter(BaseModel):
+    win_step_um: float = Field(default=50.0, description="")
+    win_sigma_um: float = Field(default=150.0, description="")
+    margin_um: float = Field(default=0.0, description="")
+    win_shape: str = Field(default="rect", description="")
+
+
+class MCInterpolateMotionKwargs(BaseModel):
+    direction: int = Field(default=1, description="0 | 1 | 2. Dimension along which channel_locations are shifted (0 - x, 1 - y, 2 - z).")
+    border_mode: str = Field(default="remove_channels", description="'remove_channels' | 'force_extrapolate' | 'force_zeros'. Control how channels are handled on border.")
+    spatial_interpolation_method: str = Field(default="idw", description="The spatial interpolation method used to interpolate the channel locations.")
+    sigma_um: float = Field(default=20.0, description="Used in the 'kriging' formula")
+    p: int = Field(default=1, description="Used in the 'kriging' formula")
+    num_closest: int = Field(default=3, description="Number of closest channels used by 'idw' method for interpolation.")
+
+
+class MCNonrigidAccurate(BaseModel):
+    detect_kwargs: MCDetectKwargs = Field(default=MCDetectKwargs(), description="")
+    localize_peaks_kwargs: MCLocalizeMonopolarTriangulation = Field(default=MCLocalizeMonopolarTriangulation(), description="")
+    estimate_motion_kwargs: MCEstimateMotionDecentralized = Field(default=MCEstimateMotionDecentralized(), description="")
+    interpolate_motion_kwargs: MCInterpolateMotionKwargs = Field(default=MCInterpolateMotionKwargs(), description="")
+
+
+class MCRigidFast(BaseModel):
+    detect_kwargs: MCDetectKwargs = Field(default=MCDetectKwargs(), description="")
+    localize_peaks_kwargs: MCLocalizeCenterOfMass = Field(default=MCLocalizeCenterOfMass(), description="")
+    estimate_motion_kwargs: MCEstimateMotionDecentralized = Field(default=MCEstimateMotionDecentralized(bin_duration_s=10.0, rigid=True), description="")
+    interpolate_motion_kwargs: MCInterpolateMotionKwargs = Field(default=MCInterpolateMotionKwargs(), description="")
+
+
+class MCKilosortLike(BaseModel):
+    detect_kwargs: MCDetectKwargs = Field(default=MCDetectKwargs(), description="")
+    localize_peaks_kwargs: MCLocalizeGridConvolution = Field(default=MCLocalizeGridConvolution(), description="")
+    estimate_motion_kwargs: MCEstimateMotionIterativeTemplate = Field(default=MCEstimateMotionIterativeTemplate(), description="")
+    interpolate_motion_kwargs: MCInterpolateMotionKwargs = Field(default=MCInterpolateMotionKwargs(border_mode="force_extrapolate", spatial_interpolation_method="kriging"), description="")
+
+
+class MCPreset(str, Enum):
+    nonrigid_accurate = "nonrigid_accurate"
+    rigid_fast = "rigid_fast"
+    kilosort_like = "kilosort_like"
+
+
 class MotionCorrection(BaseModel):
     compute: bool = Field(default=True, description="Whether to compute motion correction")
     apply: bool = Field(default=False, description="Whether to apply motion correction")
```

Comment:
```suggestion
    strategy: Literal["skip", "compute", "apply"] = Field(default="compute", description="What strategy to use for motion correction")
```
What do you think about this? This is what we ended up using in our pipelines ;)

Reply:
@alejoe91 what is the difference between compute and apply? Is there any case where we would like to compute but to not apply?

Reply:
yes, you might want to leave motion correction to KS, but still compute it externally for visualization or comparison

Reply:
sounds good then